ingestion/src/metadata/ingestion/source/deltalake.py

```diff
@@ -55,24 +64,61 @@ def __init__(
             service_name=config.service_name,
         )
         self.status = SQLSourceStatus()
-        # spark session needs to initiated outside the workflow and pass it through WorkflowContext
-        self.spark = ctx.spark
+        logger.info("Establishing Sparks Session")
+        builder = (
+            pyspark.sql.SparkSession.builder.appName(self.config.app_name)
+            .enableHiveSupport()
+            .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
+            .config(
+                "spark.sql.catalog.spark_catalog",
+                "org.apache.spark.sql.delta.catalog.DeltaCatalog",
+            )
+        )
+        if self.config.metastore_host_port:
+            builder.config(
+                "hive.metastore.uris", f"thrift://{self.config.metastore_host_port}"
+            )
+        elif self.config.metastore_file_path:
+            builder.config(
+                "spark.sql.warehouse.dir", f"{self.config.metastore_file_path}"
+            )
+        self.spark = configure_spark_with_delta_pip(builder).getOrCreate()
+
+    def set_spark(self, spark):
+        self.spark = spark
 
     @classmethod
     def create(
         cls, config_dict: dict, metadata_config_dict: dict, ctx: WorkflowContext
     ):
-        config = DeltaLakeSourceConfig.parse_obj(config_dict)
+        config = DeltalakeSourceConfig.parse_obj(config_dict)
         metadata_config = MetadataServerConfig.parse_obj(metadata_config_dict)
         return cls(config, metadata_config, ctx)
 
     def next_record(self) -> Iterable[OMetaDatabaseAndTable]:
         schemas = self.spark.catalog.listDatabases()
         for schema in schemas:
-            if not self.config.schema_filter_pattern.included(schema):
-                self.status.filter(schema, "Schema pattern not allowed")
+            if not self.config.schema_filter_pattern.included(schema.name):
+                self.status.filter(schema.name, "Schema pattern not allowed")
                 continue
-            yield from self.fetch_tables(schema)
+            yield from self.fetch_tables(schema.name)
+
+    def get_status(self):
+        return self.status
+
+    def prepare(self):
+        return super().prepare()
+
+    def _get_table_type(self, table_type):
+        if table_type.lower() == TableType.External.value.lower():
+            return TableType.External.value
+        elif table_type.lower() == TableType.View.value.lower():
```

Comment:
No need for `elif` as we are already using `return`. This can be a chain of `if - return` which makes it cleaner

Reply:
Actually, this is no bueno. Let's use a dictionary mapping the input-output of the logic:

```python
table_type_map = {
    TableType.External.value.lower(): TableType.External.value,
    ...
}

def _get_table_type(...):
    return table_type_map.get(table_type.lower(), TableType.Regular.value)
```